udata/tests/test_search.py

```diff
@@ -541,6 +553,39 @@ def test_query_with_both_including_and_excluding_terms(self):
         }
         self.assert_dict_equal(get_query(search_query), expected)
 
+    def test_query_with_multiple_including_and_excluding_terms(self):
+        '''A query should detect negation on each term in query_string'''
+        search_query = search.search_for(FakeSearch,
+                                         q='test -negated value -other')
+        expected = {
+            'bool': {
+                'must': [
+                    {'multi_match': {
+                        'query': 'test value',
+                        'analyzer': search.i18n_analyzer._name,
+                        'type': 'cross_fields',
+                        'operator': 'and',
+                        'fields': ['title^2', 'description']
+                    }}
+                ],
+                'must_not': [
+                    {'multi_match': {
+                        'query': 'negated',
```

Comment:
What do you think about **first_negated** / **1st_negated** and **second_negated** / **2nd_negated** or **other_negated** as query's words?